cartographer_ros/cartographer_ros/map_builder_bridge.cc

```diff
@@ -206,13 +207,19 @@ visualization_msgs::MarkerArray MapBuilderBridge::GetTrajectoryNodesList() {
   const auto trajectory_nodes =
       map_builder_.sparse_pose_graph()->GetTrajectoryNodes();
   int marker_id = 0;
-  for (const auto& single_trajectory : trajectory_nodes) {
+  for (int trajectory_id = 0; trajectory_id < (int)trajectory_nodes.size();
```

Comment:
Since trajectory_id is always declared an int in the rest of the same file, I'll use a static cast.